File: src/main/java/org/example/kalkulationsprogramm/dto/Mitarbeiter/MitarbeiterErstellenDto.java
package org.example.kalkulationsprogramm.dto.Mitarbeiter;

import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Data
public class MitarbeiterErstellenDto {
    private String vorname;
    private String nachname;
    private String strasse;
    private String plz;
    private String ort;
    private String email;
    private String telefon;
    private String festnetz;
    private String qualifikation;
    private BigDecimal stundenlohn;
    private Integer jahresUrlaub;
    private LocalDate geburtstag;
    private LocalDate eintrittsdatum;
    private Boolean aktiv;
    private List<Long> abteilungIds;
    private String beschaeftigungsart;
    private Long krankenkasseId;
    private Boolean kinderlos;
    private Boolean istGeschaeftsfuehrer;
    private BigDecimal kalkulatorischerLohnMonat;
    private BigDecimal geldwertVorteilMonat;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/dto/Mitarbeiter/MitarbeiterErstellenDto.kt
package org.example.kalkulationsprogramm.dto.Mitarbeiter

import java.math.BigDecimal
import java.time.LocalDate

data class MitarbeiterErstellenDto(
    var vorname: String? = null,
    var nachname: String? = null,
    var strasse: String? = null,
    var plz: String? = null,
    var ort: String? = null,
    var email: String? = null,
    var telefon: String? = null,
    var festnetz: String? = null,
    var qualifikation: String? = null,
    var stundenlohn: BigDecimal? = null,
    var jahresUrlaub: Int? = null,
    var geburtstag: LocalDate? = null,
    var eintrittsdatum: LocalDate? = null,
    var aktiv: Boolean? = null,
    var abteilungIds: List<Long>? = null,
    var beschaeftigungsart: String? = null,
    var krankenkasseId: Long? = null,
    var kinderlos: Boolean? = null,
    var istGeschaeftsfuehrer: Boolean? = null,
    var kalkulatorischerLohnMonat: BigDecimal? = null,
    var geldwertVorteilMonat: BigDecimal? = null,
)
